sklearn/tree/_tree.pyx

```diff
@@ -217,8 +214,7 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                     impurity = splitter.node_impurity()
                     first = 0
 
-                is_leaf = (is_leaf or
```

Comment:
Could you please add a quick inline comment to explain the issue with rounding errors?